hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1637,8 +1637,6 @@ protected void setWriteSchemaForDeletes(HoodieTableMetaClient metaClient) {
         } else {
           throw new HoodieIOException("Latest commit does not have any schema in commit metadata");
         }
-      } else {
-        throw new HoodieIOException("Deletes issued without any prior commits");
       }
```

Comment:
Shouldn't we throw exception if there is no dataset to delete ?

Reply:
The user doesn't care if the dataset exists, No exceptions should be thrown

Reply:
I'm, wondering what the behavior is if we delete from an empty MySQL or PostgresSQL table then, we better keep in line with the RDBMS semantics.

Reply:
![image](https://user-images.githubusercontent.com/36446084/211499510-ed715efc-cb6a-4b21-b0dd-54174c51f03a.png)
use mysql delete data ;
DELETE from t_ds_k8s  where id = 20
return information follower:
Updated Rows	0
QUERY DELETE from t_ds_k8s  where id = 20;
t_ds_k8s is empty table .No exceptions are thrown

Reply:
So I think hudi should remain semantic

Reply:
Make sense.